src/pykeen/training/training_loop.py

```diff
@@ -179,22 +203,39 @@ def train(
         # In some cases, e.g. using Optuna for HPO, the cuda cache from a previous run is not cleared
         torch.cuda.empty_cache()
 
-        result = self._train(
-            num_epochs=num_epochs,
-            batch_size=batch_size,
-            slice_size=slice_size,
-            label_smoothing=label_smoothing,
-            sampler=sampler,
-            continue_training=continue_training,
-            only_size_probing=only_size_probing,
-            use_tqdm=use_tqdm,
-            use_tqdm_batch=use_tqdm_batch,
-            tqdm_kwargs=tqdm_kwargs,
-            stopper=stopper,
-            result_tracker=result_tracker,
-            sub_batch_size=sub_batch_size,
-            num_workers=num_workers,
-        )
+        # If a checkpoint file is given we check whether it exists already and load it, if it does
+        if checkpoint_file:
+            if os.path.isfile(checkpoint_file):
```

Comment:
Likely just a matter of taste, but I prefer the high-level [pathlib](https://docs.python.org/3/library/pathlib.html?highlight=pathlib#module-pathlib) for working with paths. It also adds a type for paths making it clear from type annotations where a path is expected (instead of the generic `str`).

Reply:
but we don't want to make users import pathlib and instantiate pathlib.Path objects

Reply:
sure, but everything internally handled could use them

Reply:
maybe we could switch that all over in a super PR

Reply:
@mberr 
Do you want to open a feature/enhancement issue?

Reply:
#128